astropy/modeling/tests/test_compound.py

```diff
@@ -543,175 +282,72 @@ def test_identity_input():
     model = ident1 & shift | rotation
     assert_allclose(model(1, 2), [-3.0, 1.0])
 
-    # Same test case but using class composition
-    TestModel = ident1 & Shift | Rotation2D
-    model = TestModel(offset_1=1, angle_2=90)
-    assert_allclose(model(1, 2), [-3.0, 1.0])
-
 
-def test_slicing_on_instances_2():
+def test_invalid_operands():
     """
-    More slicing tests.
-
-    Regression test for https://github.com/embray/astropy/pull/10
+    Test that certain operators do not work with models whose inputs/outputs do
+    not match up correctly.
     """
 
-    model_a = Shift(1, name='a')
-    model_b = Shift(2, name='b')
-    model_c = Rotation2D(3, name='c')
-    model_d = Scale(2, name='d')
-    model_e = Scale(3, name='e')
-
-    m = (model_a & model_b) | model_c | (model_d & model_e)
-
     with pytest.raises(ModelDefinitionError):
-        # The slice can't actually be taken since the resulting model cannot be
-        # evaluated
-        assert m[1:].submodel_names == ('b', 'c', 'd', 'e')
-
-    assert m[:].submodel_names == ('a', 'b', 'c', 'd', 'e')
-    assert m['a':].submodel_names == ('a', 'b', 'c', 'd', 'e')
-
-    with pytest.raises(ModelDefinitionError):
-        assert m['c':'d'].submodel_names == ('c', 'd')
-
-    assert m[1:2].name == 'b'
-    assert m[2:7].submodel_names == ('c', 'd', 'e')
-    with pytest.raises(IndexError):
-        m['x']
-    with pytest.raises(IndexError):
-        m['a': 'r']
-
-    with pytest.raises(ModelDefinitionError):
-        assert m[-4:4].submodel_names == ('b', 'c', 'd')
+        Rotation2D(90) | Gaussian1D(1, 0, 0.1)
 
     with pytest.raises(ModelDefinitionError):
-        assert m[-4:-2].submodel_names == ('b', 'c')
-
-
-def test_slicing_on_instances_3():
```

Comment:
This test should be kept: some of it works, some fails because of the slicing issues.